sql/src/main/java/io/crate/operation/aggregation/impl/SumAggregation.java

```diff
@@ -31,57 +31,103 @@
 import io.crate.types.DataTypes;
 import org.elasticsearch.common.breaker.CircuitBreakingException;
 
+import javax.annotation.Nullable;
 import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.function.BiFunction;
 
-public class SumAggregation extends AggregationFunction<Double, Double> {
+public class SumAggregation<T extends Number> extends AggregationFunction<T, T> {
 
     public static final String NAME = "sum";
 
     private final FunctionInfo info;
+    private final BiFunction<T, T, T> addition;
+    private final DataType<T> type;
+    private final int bytesSize;
+
+    public static final Map<DataType, SumAggregation> sumFunctions;
+
+    static {
+        Map<DataType, SumAggregation> m = new HashMap<>();
+
+        m.put(DataTypes.FLOAT, new SumAggregation<Float>(DataTypes.FLOAT, (n1, n2) -> n1 + n2));
+        m.put(DataTypes.DOUBLE, new SumAggregation<Double>(DataTypes.DOUBLE, (n1, n2) -> n1 + n2));
+
+        final BiFunction<Long, Long, Long> longAddImpl = (n1, n2) -> {
+            Long r = n1 + n2;
+            // HD 2-12 Overflow iff both arguments have the opposite sign of the result
```

Comment:
this code can be found when looking at Math.addExact(long x, long y)
i was asked to use the implementation without the primitive type for performance increase.
but you are right, the HD 2-12 is unnecessary.